frontend/src/stores/courses.js

```diff
@@ -0,0 +1,126 @@
+import { decorate, observable, action, autorun } from 'mobx';
+import mockCourse from './course-mock.json';
+import dateFns from 'date-fns';
+
+const isAvailableByTime = (courseItem) =>
+    // within 3 days from now
+    dateFns.differenceInDays(courseItem.startDate, new Date()) < 3 &&
+    // and must not be 1 hours before starting time
+    dateFns.differenceInHours(courseItem.startDate, new Date()) > 1;
+
+const pluckId = (obj) => {
+    const clone = [];
```

Comment:
This one is a little hard to follow. Could you elaborate the expected input with a comment. We do usually practice functional style rather than the implicit as it is a little cleaner. With functional stuff it would look like this:

```
const pluckId = (obj) => Object.values(obj).map(arr => arr.map(item => item.id)).reduce((result, ids) => [...result, ...ids], []);
```

Reply:
Anyway, you could use the Object.values to skip the first `forEach` and the function name could also be reflecting the plural functionality of it so the name would be `pluckIds`.